src/Yaapii.Atoms/Map/Map.Envelope.cs

```diff
@@ -37,22 +37,41 @@ public partial class Map
         public abstract class Envelope : IDictionary<string, string>
         {
             private readonly UnsupportedOperationException rejectWriteExc = new UnsupportedOperationException("Writing is not supported, it's a read-only map");
-
             private readonly Func<IDictionary<string, string>> origin;
             private readonly Sticky<IDictionary<string, string>> fixedOrigin;
             private readonly bool live;
+            private readonly Func<string, string> fallback;
 
             /// <summary>
             /// Simplified map building.
             /// </summary>
-            public Envelope(Func<IDictionary<string, string>> origin, bool live)
+
+            public Envelope(Func<IDictionary<string, string>> origin, bool live) : this(
+              origin, live,
+              key => throw new ArgumentException($"The key '{key}' is not present in the map.")
+            )
+            { }
+
+            public Envelope(Func<IDictionary<string, string>> origin, bool live, Func<string, string> fallback)
```

Comment:
FallBack map should be realized by making a decorator. Envelopes are only for simplifying making classes for types. An envelope should only delegate the methods to the injected instance of the type. It may have default behaviour which we want for all classes, but thats it. Having a fallback, however, is a specific behaviour.

Reply:
Sorry for not pointing this out before, I did not look close enough at your solution.

Reply:
As I used a different approach I started again on a different branch and created a new PR #373